File: app/src/main/java/ivankuo/com/itbon2018/di/ActivityBuildersModule.java
package ivankuo.com.itbon2018.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ivankuo.com.itbon2018.MainActivity;

@Module
public abstract class ActivityBuildersModule {
    @ContributesAndroidInjector(modules = FragmentBuildersModule.class)
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/ivankuo/com/itbon2018/di/ActivityBuildersModule.kt
package ivankuo.com.itbon2018.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import ivankuo.com.itbon2018.MainActivity

@Module
abstract class ActivityBuildersModule {
    @ContributesAndroidInjector(modules = arrayOf(FragmentBuildersModule::class))
    abstract fun contributeMainActivity(): MainActivity
}